File: slack-reader/src/main/java/dev/number6/slackreader/model/WorkspaceMessages.java
package dev.number6.slackreader.model;

import java.time.LocalDate;
import java.util.Collection;
import java.util.HashMap;
import java.util.Map;
import java.util.Set;
import java.util.stream.Collectors;

public class WorkspaceMessages {
    private final LocalDate comprehensionDate;
    private final Map<String, Collection<String>> channelMessages;

    public WorkspaceMessages() {
        this(LocalDate.now());
    }

    public WorkspaceMessages(LocalDate comprehensionDate) {
        this.channelMessages = new HashMap<>();
        this.comprehensionDate = comprehensionDate;
    }

    public void add(String channelName, Collection<String> messages) {
        this.channelMessages.put(channelName, messages);
    }

    public LocalDate getComprehensionDate() {
        return comprehensionDate;
    }

    public Set<String> getChannelNames() {
        return channelMessages.keySet();
    }

    public Collection<String> getMessagesForChannel(String channelName) {
        return channelMessages.get(channelName);
    }

    public Set<String> getActiveChannelNames() {
        return getChannelNames().stream().filter(c -> getMessagesForChannel(c).size() > 0).collect(Collectors.toSet());
    }
}

File: slack-reader/src/main/kotlin/dev/number6/slackreader/model/WorkspaceMessages.kt
package dev.number6.slackreader.model


import java.time.LocalDate
import java.util.*
import java.util.stream.Collectors
import kotlin.collections.ArrayList

class WorkspaceMessages @JvmOverloads constructor(comprehensionDate: LocalDate = LocalDate.now()) {
    val comprehensionDate: LocalDate
    private val channelMessages: MutableMap<String, Collection<String?>>
    fun add(channelName: String, messages: Collection<String?>): WorkspaceMessages {
        channelMessages[channelName] = messages
        return this
    }

//    fun getComprehensionDate(): LocalDate {
//        return comprehensionDate
//    }

    private fun getChannelNames(): Set<String> {
        return channelMessages.keys
    }

    fun getMessagesForChannel(channelName: String): Collection<String?> {
        return channelMessages[channelName] ?: ArrayList()
    }

    fun getActiveChannelNames(): Set<String> {
        return getChannelNames().stream().filter { c -> getMessagesForChannel(c).isNotEmpty() }.collect(Collectors.toSet())
    }

    init {
        channelMessages = HashMap()
        this.comprehensionDate = comprehensionDate
    }
}